lisa/analysis/rta.py

```diff
@@ -40,7 +40,7 @@ class PerfAnalysis(AnalysisHelpers):
     RTA_LOG_PATTERN = 'rt-app-{task}.log'
     "Filename pattern matching RTApp log files"
 
-    def __init__(self, task_log_map):
+    def __init__(self, task_log_map, bundle=None):
```

Comment:
analysis classes are independent from TestBundle and only rely on traces or in that case rtapp logs. Creating the analysis from the bundle res dir is straightforward already:
```
perf_analysis = PerfAnalysis.from_dir(bundle.res_dir)
```

Below functions need not be aware of test bundles. It may be acceptable to add a method in RTATestBundle to return a PerfAnalysis, but only if it proves to add anything useful beyond the above one liner.

Reply:
You right, we can use the TestBundle as a factory for the PerfAnalysis... it actually looks even better.
Thanks for the suggestion!